src/ILLink.RoslynAnalyzer/AnalyzerUtils.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+using static ILLink.RoslynAnalyzer.RequiresUnreferencedCodeAnalyzer;
+
+namespace ILLink.RoslynAnalyzer
+{
+	static class AnalyzerUtils
+	{
+		/// <summary>
+		/// Returns true if <see paramref="type" /> has the same name as <see paramref="typename" />
+		/// </summary>
+		internal static bool HasName (this INamedTypeSymbol type, string typeName)
+		{
+			var roSpan = typeName.AsSpan ();
+			INamespaceOrTypeSymbol? currentType = type;
+			while (roSpan.Length > 0) {
+				var dot = roSpan.LastIndexOf ('.');
+				var currentName = dot < 0 ? roSpan : roSpan.Slice (dot + 1);
+				if (currentType is null ||
+					!currentName.Equals (currentType.Name.AsSpan (), StringComparison.Ordinal)) {
+					return false;
+				}
+				currentType = (INamespaceOrTypeSymbol?) currentType.ContainingType ?? currentType.ContainingNamespace;
+				roSpan = roSpan.Slice (0, dot > 0 ? dot : 0);
+			}
+
+			return true;
+		}
+
+		const string SuppressMessageFqn = "System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessageAttribute";
+
+		internal static bool IsDiagnosticSuppressed(OperationAnalysisContext operationContext, string diagnosticId)
+		{
+			var attributes = operationContext.ContainingSymbol.GetAttributes();
```

Comment:
Will this work when the suppression is up in the tree?
e.g.
```C#
[UnconditionalSuppressMessage ("", "IL1234:MyWarning")]
class Foo {
   void Bar () {
     // Triggers warning IL1234
  }
}
```

Reply:
Right, this only handles local suppression. Global suppression support will have to be added in another PR.